client/src/views/SignUp.js

```diff
@@ -1,43 +1,105 @@
-import React from 'react';
+import React, {Component} from 'react';
 import '../sass/LoginSignup.scss';
 import { Button, Form, Grid, Header, Message, Segment } from 'semantic-ui-react';
+import registerUser from '../backendCalls/auth/registerUser.js';
 
-const SignUpForm = () => (
-  <div className='signup-form'>
-    <Grid textAlign='center' style={{ height: '100%' }} verticalAlign='middle'>
-      <Grid.Column style={{ maxWidth: 450 }}>
-        <Header as='h2' color='black' textAlign='center'>
-          Sign up
-        </Header>
-        <Form size='large'>
-          <Segment raise>
-            <Form.Input fluid icon='user' iconPosition='left' placeholder='Username/email' />
-            <Form.Input fluid icon='mail' iconPosition='left' placeholder='Email' />
-            <Form.Input
-              fluid
-              icon='key'
-              iconPosition='left'
-              placeholder='Password'
-              type='password'
-            />
-            <Form.Input
-              fluid
-              icon='key'
-              iconPosition='left'
-              placeholder='Repeat password'
-              type='password'
-            />
-            <Button color='blue' fluid size='large'>
-              Create your account
-            </Button>
-          </Segment>
-        </Form>
-        <Message>
-          Already a member? <a href='/login'>Log in</a>
-        </Message>
-      </Grid.Column>
-    </Grid>
-  </div>
-)
+class SignUpForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formData: new Map([
+        ['username', ''],
+        ['email', ''],
+        ['password', ''],
+        ['repeatPassword', '']
+      ]),
+      result: ''
+    };
+  }
 
-export default SignUpForm;
\ No newline at end of file
+  handleDataChange = (e, item) => {
+    let newData = this.state.formData;
+    newData[item] = e.target.value;
+    this.setState({formData: newData});
+  };
+
+  handleSubmit = async(e) => {
+    let data = this.state.formData;
+      if (data['password'] !== data['repeatPassword']) {
+        this.setState({result: <font color='#ff0000'> Passwords do not match</font>})
+
+      } else {
+        let response = await registerUser(data['email'], data['password'], data['username'], undefined);
+        let responseHTML = '';
+        if (response === 200) {
+          responseHTML = 'Account successfully created';
+        } else {
+          responseHTML = <font color='#ff0000'> {response} </font>;
```

Comment:
I saw that you set `color='blue'` on a button in another part of the code, is it possible to do `color='red'` here?